projects/js-toolkit/packages/js-toolkit-core/src/template/Renderer.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import ejs from 'ejs';
+import fs from 'fs-extra';
+
+import FilePath from '../file/FilePath';
+
+/**
+ * Template renderer class
+ */
+export default class Renderer {
+	constructor(templatesDir: FilePath, outputDir?: FilePath) {
+		this._templatesDir = templatesDir;
+		this._outputDir = outputDir;
+	}
+
+	/**
+	 * @param templateRelPosixPath the template path (relative to templates dir) in POSIX format
+	 * @param data the contextual data to render the template
+	 * @return the contents of the rendered template
+	 */
+	render(templateRelPosixPath: string, data: object = {}): Promise<string> {
```

Comment:
`templateRelPosixPath` => `templateRelativePosixPath`?

Reply:
See comment above. I used a naming convention and the `FilePath` class for paths because of how error prone it is path handling in a tool like this :scream: .